bpf/polprog/pol_prog_builder.go

```diff
@@ -205,6 +208,30 @@ func (p *Builder) setUpIPSetKey(ipsetID uint64, keyOffset, ipOffset, portOffset
 	p.b.StoreStack32(R1, keyOffset+ipsKeyID+4)
 }
 
+func validateNet (rule *proto.Rule) (*proto.Rule) {
```

Comment:
This formatting looks wrong, you should do `make install-git-hooks` to add the pre-commit check hooks and run `make fix` to format the code before check-in.

Reply:
I think you should 

* Move this function over to the rules package so it can be used from both places; there's still a lot of duplication between this and `ProtoRuleToIptablesRules`.
* Move the check of rule.IpVersion into the start of this function.  (i.e. return nil if the rule doesn't match the IP version that we're rendering for.)

Also, I think the naming could be improved.  How about `FilterRuleToIPVersion(rule, ipVersion)`?

Reply:
ipversion check for policy includes v6 as well right? For BPF, we dont want to go ahead with policy writing if its v6. I feel ipversion check inside this function may not be possible until we support v6 in bpf mode.

Reply:
Can have additional flags and check if its BPF mode or not. your comments?

Reply:
Here's what I'd do: pass in the IP version that you're rendering for:
```
func FilterRuleToIPVersion(ipVersion int, rule *proto.Rule) *proto.Rule {
    if rule.IpVersion != 0 && rule.IpVersion != ipVersion {
        log.Debugf ("Skipping rule it is for a different IP version.")
        return nil
    }

    ...
}
```

Then, in the BPF code, pass in the static `ipVersion=4`.  In the iptables-mode code, pass in the `ipVersion` variable that contains the version that we're rendering for.